mne/channels/layout.py

```diff
@@ -163,18 +163,30 @@ def _read_lay(fname):
     return box, pos, names, ids
 
 
-def read_layout(kind, path=None, scale=True):
+def read_layout(kind=None, path=None, fname=None, *, scale=True):
```

Comment:
1. All existing user code currently will have either `read_layout(name, ...)` or `read_layout(kind=name, ...)`. It would be nice if at least the first one (which might even be the most common!) continued to work without a deprecation warning if possible. So what if we deprecate differently by doing:
    ```suggestion
    def read_layout(fname=None, path=None, scale=True, *, kind=None):
        if kind is not None:
            warn(...)
            ...
        if path is not None:
            warn(...)
            ...
    ```
    That way `read_layout('Vectorview-all')` works on 1.3 and also in this PR without any warning.

2. Since the current default for `path` is `None`, we need to use something other than `None` as the default now to know if a user has actually passed the param or not. otherwise code like this:
    ```
    read_layout('VectorView-all', path=None)
   ```
    works on 1.3, does not warn the user for passing `path=None` in 1.4, yet will raise an error about the kwarg not existing come 1.5. We could for example use `path=''` (I think it's safe to assume no user has done `path=''` in their code) and do:
    ```
    _validate_type(path, ('path-like', None), 'path')
    if path != '':
        warn(...)
    ```
    as our user-passed-so-need-to-warn check.

Reply:
Good points, it should be better now, thanks!